src/infra/core/src/testing/mock_dataset_action_authorizer.rs

```diff
@@ -78,6 +116,22 @@ impl MockDatasetActionAuthorizer {
             .expect_check_action_allowed()
             .returning(|_, _| Ok(()));
         mock_dataset_action_authorizer
+            .expect_classify_dataset_ids_by_allowance()
+            .returning(|ids, _| {
+                Ok(ClassifyByAllowanceIdsResponse {
+                    authorized_ids: ids,
+                    unauthorized_ids_with_errors: Vec::new(),
+                })
+            });
+        mock_dataset_action_authorizer
```

Comment:
Notice: `mockall` mocks panic if _none_ of the `expect_*`ed methods was called. After I changed `QueryService` to use batch `classify_dataset_ids_by_allowance` methods the tests started crashing.

To solve this, in `MockDatasetActionAuthorizer::allowing()` and `denying()` I add `exect`s for both singular and batch methods, and if at least one of them is called - the mock will be happy.

Reply:
@zaychenko-sergei I will go ahead an merge this to propagate changes to the node faster, but wanted to point out this specific change to you, as I saw a lot of complexity forming in this mock class.